app/renderer/abr-document.js

```diff
@@ -251,6 +252,35 @@ AbrDocument.prototype = {
         return this.cm.doc.setValue("");
     },
 
+    // Get document title suggestion for save-dialog
+    getTitleSuggestion: function() {
+        var data = this.getData();
```

Comment:
I think it is much better in term of performances to use CodeMirror methods:

* `doc.firstLine()` or `doc.getLine(0)` to get the first line,
* `doc.eachLine()` to iterate over the document lines.

See http://codemirror.net/doc/manual.html#api_content

With this approach you won't need to manipulate the whole `data` and to use `os.EOL` etc.